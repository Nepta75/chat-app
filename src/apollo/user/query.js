import { gql } from '@apollo/client';

export const userQuery = gql`
  query user {
    user @client
  }
`;