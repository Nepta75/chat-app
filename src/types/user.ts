export interface User {
  id: number
  nickname: string
  token: string
  password: string
}