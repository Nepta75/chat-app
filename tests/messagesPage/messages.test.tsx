import renderer, { act } from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import Messages from "../../src/pages/index";
import { userQuery } from '../../src/apollo/user/query';
import { InMemoryCache } from '@apollo/client';
import { User } from '../../src/types/user'
import { Conversation } from '../../src/types/conversation'

const user: User = {
  "id": 1,
  "nickname": "Thibaut",
  "password": "123",
  "token": "xxxx"
};

const cache = new InMemoryCache().restore({
  ROOT_QUERY: {
    user,
  },
});

const MOCK_CONVERSATIONS: Array<Conversation> =  [
  {
    "id": 1,
    "recipientId": 2,
    "recipientNickname": "Jeremie",
    "senderId": 1,
    "senderNickname": "Thibaut",
    "lastMessageTimestamp": 1625637849
  },
  {
    "id": 2,
    "recipientId": 3,
    "recipientNickname": "Patrick",
    "senderId": 1,
    "senderNickname": "Thibaut",
    "lastMessageTimestamp": 1620284667
  },
  {
    "id": 3,
    "recipientId": 1,
    "recipientNickname": "Thibaut",
    "senderId": 4,
    "senderNickname": "Elodie",
    "lastMessageTimestamp": 1625648667
  }
];

const mocks = [
  {
    request: {
      query: userQuery,
    },
    result: { data: { user } },
  },
];

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(MOCK_CONVERSATIONS),
  }),
) as jest.Mock;

describe('message page', () => {
  test('Should render correctly with user and conversations mocked data', async () => {
    let component;
    await act(async () => {
      component = renderer.create(
        <MockedProvider cache={cache} mocks={mocks}>
          <Messages />
        </MockedProvider>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
