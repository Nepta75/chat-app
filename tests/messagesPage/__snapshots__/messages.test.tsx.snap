// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message page Should render correctly with user and conversations mocked data 1`] = `
<div
  className="container"
>
  <div
    className="Messagesstyle__StyledContainer-sc-15ymc2e-0 fPldhQ"
  >
    <div
      className="conversations-container"
    >
      <div
        className="item "
        onClick={[Function]}
      >
        <img
          alt="recipient-2"
          src="https://www.wortis.fr/wp-content/uploads/2018/06/client.png"
        />
        <div>
          Jeremie
        </div>
        <div
          className="date"
        >
          dernier message le 
          19/01/70 à 08:33
        </div>
      </div>
      <div
        className="item "
        onClick={[Function]}
      >
        <img
          alt="recipient-3"
          src="https://www.wortis.fr/wp-content/uploads/2018/06/client.png"
        />
        <div>
          Patrick
        </div>
        <div
          className="date"
        >
          dernier message le 
          19/01/70 à 07:04
        </div>
      </div>
      <div
        className="item "
        onClick={[Function]}
      >
        <img
          alt="recipient-1"
          src="https://www.wortis.fr/wp-content/uploads/2018/06/client.png"
        />
        <div>
          Elodie
        </div>
        <div
          className="date"
        >
          dernier message le 
          19/01/70 à 08:34
        </div>
      </div>
    </div>
  </div>
</div>
`;
